### ISSUE TITLE
How to Set Scheduled Power On/OFF?

### ISSUE THREAD

**Issue by AP0rPi7uQ9** [phhusson/treble_experimentations#2445]
I used miui before and I can set scheduled Power On/OFF.

When i go to bed at 11:00 pm, the phone automatically power off.
When I get up at 7:00 am, the phone automatically power on.

How should i set it up in AOSP?
In settings i didn't find any option about scheduled power ON/OFF

I use AOSP 12.1 v414 system-squeak-arm64-ab-gapps.img.xz, My phone is Redmi K40(alioth)

By the way, When I set an alarm clock and I manually power off, the alarm clock in the power off state does not automatically power on and ring like miui. How should I set up to achieve this functionality?

**Comment by phhusson** [comment]
@AP0rPi7uQ9 The short answer is that Google, thus AOSP, doesn't support this feature, so it would require us to make the full integration of the feature, it wouldn't /just/ be connecting/fixing the drivers together, so that's quite a lot of work, and it wouldn't be standard: It needs the Clock app to be specifically modified for that, and non-AOSP ROMs will probably want to use other Clock apps.

So basically, Google doesn't even know this is possible (even though I'm pretty confident Pixels could do it as well), and this makes it pretty painful for us to implement.

**Comment by AP0rPi7uQ9** [comment]
@phhusson @sd369888 
thanks, thank you for your explain, This is my first time to use GSI & aosp . So Im not very clear about it.

By the way,May I know , is there any difference between your aosp gsi(treble expermitation) and the google gsi of google's official website?

**Comment by AP0rPi7uQ9** [comment]
thank
